### PR TITLE
shaderc: new port

### PR DESCRIPTION
**Pull request by jasonliu--**
#### Description

[Shaderc](https://github.com/google/shaderc) is a collection of tools, libraries, and tests for shader compilation. It includes glslc, a command line compiler for GLSL/HLSL to SPIR-V, and libshaderc, a library API for accessing glslc functionality.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G14042 x86_64
Xcode 10.1 10B61

macOS 10.11.6 15G22010 x86_64
Xcode 8.2.1 8C1002

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by jasonliu--** [comment]
The CI build failed with the following error messages:

```
Error: Failed to patch shaderc: error deleting
    "/opt/local/var/macports/home/Library/Preferences/com.apple.dt.Xcode.plist": permission denied
    while executing
"file delete -force "${target_dir}/com.apple.dt.Xcode.plist""
    (procedure "macports::copy_xcode_plist" line 5)
    invoked from within
"macports::copy_xcode_plist $env(HOME)"
    (procedure "mportinit" line 646)
    invoked from within
"mportinit ui_options global_options global_variations"
Error: /opt/local/bin/port: Failed to initialize MacPorts, error deleting
    "/opt/local/var/macports/home/Library/Preferences/com.apple.dt.Xcode.plist": permission denied
DEBUG: Error code: NONE
DEBUG: Backtrace: error deleting
    "/opt/local/var/macports/home/Library/Preferences/com.apple.dt.Xcode.plist": permission denied
    while executing
"file delete -force "${target_dir}/com.apple.dt.Xcode.plist""
    (procedure "macports::copy_xcode_plist" line 5)
    invoked from within
"macports::copy_xcode_plist $env(HOME)"
    (procedure "mportinit" line 646)
    invoked from within
"mportinit ui_options global_options global_variations"
Error: /opt/local/bin/port: Failed to initialize MacPorts, error deleting
    "/opt/local/var/macports/home/Library/Preferences/com.apple.dt.Xcode.plist": permission denied
    while executing
"$post $targetname"
```

Not quite sure what the problem is, but this isn't happening on my own machines.

File: graphics/shaderc/Portfile [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    reinplace "/Update build-version.inc/a\\
+        add_custom_target(build-version\\
+        \\  COMMAND \${CMAKE_COMMAND} -E copy\\
+        \\          \${CMAKE_SOURCE_DIR}/build-version.inc\\
+        \\          \${CMAKE_CURRENT_BINARY_DIR}/build-version.inc\\
+        )\\
+    " \
+        ${worksrcpath}/CMakeLists.txt
+    generate_versions_file
+}
```
Comment (reneeotten) [inline graphics/shaderc/Portfile:106]: much, if not all, of the reinplace stuff above and below if this comment should be done as patches; no e of this is readable and maintainable long-term

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: graphics/shaderc/Portfile [hunk 2]
```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+github.setup        google shaderc 2023.7 v
+revision            0
+
```
[the remark this answers is not in the document]
Comment (jasonliu--) [reply graphics/shaderc/Portfile:10]: Added.

File: graphics/shaderc/Portfile (hunk 1 above)
Comment (jasonliu--) [reply graphics/shaderc/Portfile:106]: Whether or not it's readable or maintainable is for me to decide, since this port is maintained by me, and isn't `openmaintainer`.

Comment (reneeotten) [reply graphics/shaderc/Portfile:106]: not completely true; patchfiles are preferred for anything besides  a simple "reinplace". Feel free to keep it like this and find someone  else to merge this. I cannot follow what's happening with all this patching/reinplacing and other non-default things that are happening in this Portfile and will thus not merge this PR as-is.

**Comment by reneeotten** [comment]
@jasonliu-- I see that my review was requested here. Unfortunately my comment regarding all the non-default things with `procs` and reinplaces that happen in this Port makes it very hard to follow what's actually happening. So as I said before, please follow the usual guidelines of using patchfiles instead. If you're interested in doing so, we would welcome the contribution; if not, then I will personally not merge this PR as I told you before. If there is no activity on this PR in the next week or so, I will close it until you have found the time to work on it more.
